packages/liveblocks-react-ui/src/primitives/Composer/index.tsx

```diff
@@ -1452,26 +1658,95 @@ const ComposerAttachmentsDropArea = forwardRef<
   }
 );
 
+/**
+ * A toggle button which toggles a specific text mark.
+ *
+ * @example
+ * <Composer.MarkToggle mark="bold">
+ *   Bold
+ * </Composer.MarkToggle>
+ */
+const ComposerMarkToggle = forwardRef<
+  HTMLButtonElement,
+  ComposerMarkToggleProps
+>(
+  (
+    {
+      children,
+      mark,
+      onValueChange,
+      onClick,
+      onPointerDown,
+      asChild,
+      ...props
+    },
+    forwardedRef
+  ) => {
+    const Component = asChild ? Slot : "button";
+    const { marks, toggleMark } = useComposer();
+
+    const handleClick = useCallback(
+      (event: MouseEvent<HTMLButtonElement>) => {
+        onClick?.(event);
+
+        if (!event.isDefaultPrevented()) {
+          toggleMark(mark);
+          onValueChange?.(mark);
+        }
+      },
+      [mark, onClick, onValueChange, toggleMark]
+    );
+
+    const handlePointerDown = useCallback(
+      (event: PointerEvent<HTMLButtonElement>) => {
+        onPointerDown?.(event);
+
+        event.preventDefault();
+        event.stopPropagation();
```

Comment:
Is there a reason why we prevent default behavior and stop propagation on pointer down?

Reply:
The floating toolbar is only opened if the editor is focused, so we prevent focus from being applied to the toolbar and its elements (for the link edition modal we won't be able to solve this in the same way because it has an input so I'm probably going to revisit this when working on links). I initially only applied this on `MarkToggle` before also adding it to `FloatingToolbar` but it makes more sense to only have it on the toolbar (since `MarkToggle` can be used elsewhere), done in https://github.com/liveblocks/liveblocks/pull/2076/commits/153b32be647f36c32dc81f61c3cf7808191e71ef.